pkg/scale/decode.go

```diff
@@ -105,15 +113,29 @@ func (d *Decoder) Decode(dst interface{}) (err error) {
 }
 
 // NewDecoder is constructor for Decoder
-func NewDecoder(r io.Reader) (d *Decoder) {
-	d = &Decoder{
-		decodeState{r},
+func NewDecoder(r io.Reader, options ...func(*Decoder)) (d *Decoder) {
+	d = &Decoder{}
+	// Default values...
+	d.decodeState.Reader = r
+	d.decodeState.maxLength = defaultMaxLength
+
+	// Option parameters values:
+	for _, option := range options {
+		option(d)
 	}
 	return
 }
 
+// OptionMaxLength sets the maximum length of the data to be decoded
+func OptionMaxLength(maxLength int) func(*Decoder) {
+	return func(options *Decoder) {
+		options.maxLength = maxLength
+	}
```

Comment:
an error would be useful here to prevent the user from settings a limit of `0` for example here.